torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1992,15 +1984,19 @@ void ONNXShapeTypeInference(
       //       The conversion here is incomplete for these ops.
       //       e.g: ListConstruct, ListUnpack, etc.
       std::shared_ptr<onnx::ModelProto> model_proto;
-      SymbolDimMap symbol_map;
-      ConvertGraphToONNXProto(n_graph, model_proto, symbol_map, opset_version);
+      ConvertGraphToONNXProto(n_graph, model_proto, symbol_dim_map, opset_version);
       GRAPH_DEBUG(
           "ONNX graph to run shape inference: ", prettyPrint(*model_proto));
 
       // infer shape
       try {
-        onnx::shape_inference::InferShapes(*model_proto);
-        UpdateOutputTypeByONNXProto(n, clone_node, *model_proto, symbol_map);
+        // TODO: add data propagation supports for more operators
+        if (n->kind() == ::c10::onnx::Shape || n->kind() == ::c10::onnx::Gather) {
+          inferred_shape_data = onnx::shape_inference::InferShapesAndDataPropagation(*model_proto, inferred_shape_data);
```

Comment:
hmm `inferred_shape_data` is both input and output. What is the right expectation? Are the `inferred_shape_data` in ConstantValueMap (incrementally) updated after this call?

Reply:
To prevent confusion, I decided to make InferShape from ONNX take `inferred_shape_data` by reference so it will be modified in place.